p2p/key.go

```diff
@@ -16,6 +16,11 @@ import (
 // NodeID is a hex-encoded crypto.Address.
 type NodeID string
 
+// NewNodeID returns a lowercased (normalized) NodeID.
+func NewNodeID(nodeID string) NodeID {
```

Comment:
This should call `Validate()` as well, to make sure it's actually valid, and `Validate()` should check that it's lowercased (since callers may not be using `NewNodeID`, e.g. if it's received via Protobuf).

We should possibly have a `Normalize()` method as well, which can be called on already-typed strings received from external sourced.

Reply:
Why would this call `Validate` if it is doing the normalizing. If anything, we should just have `NewNodeID` and `Normalize`, where the former calls the later. Also, we could have a `Validate` function too.

Reply:
Because it needs to e.g. check that it's actually a valid hex string, not just some random garbage.

`NodeID` can be embedded in structs, and these structs can be received via the network (e.g. from Protobuf handshakes), so we may want to have `Validate()` and/or `Normalize()` to handle this. I believe we also have some existing P2P code from the old stack that relies on `Validate()`.

Reply:
Ok, yeah I think having `NewNodeID` ensure input is a valid hex-string and then lower-case should suffice. We'll have a validate function as well.